src/screens/staking/lib/staking_sdk/wallet_operations/cosmos.ts

```diff
@@ -118,6 +129,136 @@ const getCosmosError = (err: Error): CosmosError => {
   );
 };
 
+type PollTxOpts = {
+  rpc: string;
+  txHash: string;
+};
+
+const pollTx = async ({ rpc, txHash }: PollTxOpts) => {
+  const tmClient = await connectComet(rpc);
+
+  const queryClient = QueryClient.withExtensions(tmClient, setupTxExtension);
+
+  while (true) {
+    try {
+      const tx = await queryClient.tx.getTx(txHash);
+
+      return tx;
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log("debug: cosmos.ts: err", err);
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 5000));
```

Comment:
No exit points in the catch block or after. Potential infinite loop?

Reply:
I didn't want to add a timeout since transactions in Injective can take a long time, but yes it is not ideal, I think I will add a 3 mins check.